Update Go deps

sethvargo opened the pull request:
This fixes a few downstream minor security issues.

myurtoglu commented:
Doesn't feel like it's a straightforward change to move from Spring Boot 2 to 3.

yolocs commented:
Sorry about the conflict

sethvargo commented:
> Doesn't feel like it's a straightforward change to move from Spring Boot 2 to 3.

I removed the Java changes - just doing Go for now.

> Sorry about the conflict

Not a problem - I moved the protos around a bit, esp talker.

yolocs commented on internal/talkerpb/talkerpb.go:
```diff
@@ -0,0 +1,17 @@
+
+//go:generate protoc -I../../third_party/googleapis -I../../internal/protos/talker --go_out=. --go-grpc_out=. --go_opt=paths=source_relative --go-grpc_opt=paths=source_relative talker.proto
+
+package talkerpb
```
It probably doesn't have any practical impact but it'd be good if we update https://github.com/abcxyz/lumberjack/blob/7b83e6469d094af974acedb1209ae7377676214f/internal/protos/talker/talker.proto#L18 to be `option go_package = "github.com/abcxyz/lumberjack/internal/talkerpb";`
